plugin/supervisor.go

```diff
@@ -103,6 +110,55 @@ func (sup *supervisor) Hooks() Hooks {
 	return sup.hooks
 }
 
+// PerformHealthCheck checks the plugin through a process check, an RPC ping, and a HealthCheck hook call.
+func (sup *supervisor) PerformHealthCheck() error {
+	if procErr := sup.CheckProcess(); procErr != nil {
+		mlog.Debug(fmt.Sprintf("Error checking plugin process, error: %s", procErr.Error()))
+		return fmt.Errorf("Plugin process not found, or not responding")
```

Comment:
Nit: You can just use `errors.New`.

Reply:
There are other occurrences of `fmt.Errorf()` without string interpolation in this file, and `errors` is not imported at the moment, which is why I went with this choice. I can change to use `errors.New` appropriately.

Reply:
2/5 that we should use 'errors.New` in this case. I don't see the reason to use `fmt.Errorf()`. It's to powerful and misleading.